src/components/UserSelector.tsx

```diff
@@ -2,15 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onSelectedUser: (val: User | null) => void;
+  selectedUser: User | null;
+  onSelectedPost: (val: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectedUser,
+  onSelectedPost,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  const handleUserSelect = (user: User) => {
+    onSelectedPost(null);
+    onSelectedUser(user);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div
+      data-cy="UserSelector"
+      className={classNames('dropdown', { 'is-active': isDropdownActive })}
+    >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsDropdownActive(true)}
```

Comment:
You are using arrow function in the `onClick` handler. It's not a mistake but it's a bad practice because it may cause unnecessary re-renders. It's better to define a method for this.